### ISSUE TITLE
Question about ID loss

### ISSUE THREAD

**Issue by Harr7y** [superhero-7/DreamID#9]
When computing the ID Embedding (stated in the paper: obtained from an off the shelf ID encoder - insightface), the input of Insightface should be in the numpy format. 
However, if we convert the generated image into numpy array (with detach() / cpu()), the graident can not be propagated back to the unet.

So I am curious about how to use the ID loss to update the weights of DreamID?

Correct me if I am wrong:
generated_image = DreamID(**input)
ID_emd = Insightface(generated_image.detach().cpu().numpy())

**Comment by superhero-7** [comment]
They have onnx model.

**Comment by ThreeSRR** [comment]
Do you use the PyTorch version of insightface to extract embeddings and compute the ID loss, instead of the onnx model?

**Comment by garychan22** [comment]
similar quesion: inference onnxruntime is not tracked by torch.auto.grad, how to backpropagte the ID loss gradient to the unet? dose anyone know?
